src/components/QueryRecordings/QueryRecordings.vue

```diff
@@ -115,246 +94,168 @@ export default {
-      this.advanced =
-        this.query.tags.length !== 0 ||
-        this.query.tagMode !== "any" ||
-        (this.query.where && this.query.where.duration.hasOwnProperty("$lte"));
-      this.canHaveTags = this.canHaveSpecifiedTags(this.query.tagMode);
-      this.animals = this.query.tags.map(tag =>
-        DefaultLabels.searchLabels().find(({ value }) => tag === value)
-      );
-    }
+    this.makeApiRequest();
   },
   methods: {
+    resetToDefaultQuery() {
+      this.selectedDevices = [];
+      this.selectedGroups = [];
+      this.dates = {
+        days: 30
+      };
+      this.duration = {};
+      this.recordingType = this.$store.state.User.recordingTypePref || "both";
+      this.tagData = {
+        tagMode: "any"
+      };
+    },
     saveLastQuery() {
-      this.lastQuery = JSON.parse(JSON.stringify(this.query));
+      this.lastQuery = this.serialiseQueryForRecall();
+      this.$emit("description", this.makeSearchDescription());
+    },
+    queryHasChanged() {
+      return (
+        JSON.stringify(this.lastQuery) !==
+        JSON.stringify(this.serialiseQueryForRecall())
+      );
     },
     updatePagination(perPage, page) {
       this.query.limit = perPage;
       const newOffset = Math.max(0, (page - 1) * perPage);
       this.query.offset = newOffset;
       this.updateRouteQuery();
+      this.makeApiRequest();
     },
-    deserialiseRouteIntoQuery(routeQuery, target) {
-      target = target || this.query;
-      for (const key in routeQuery) {
-        if (routeQuery.hasOwnProperty(key)) {
-          target[key] = routeQuery[key];
-        }
+    setOnlyIfExists(itemName, source, destination) {
+      if (source.hasOwnProperty(itemName)) {
+        destination[itemName] = source[itemName];
       }
-      if (routeQuery.where) {
-        target.where = JSON.parse(routeQuery.where);
-        if (!target.where.recordingDateTime) {
-          this.$set(target.where, "recordingDateTime", {});
-        }
-        if (!target.where.dateRange) {
-          this.$set(target.where, "dateRange", {});
-        } else {
-          if (target.where.dateRange === "customDateRange") {
-            this.$set(target.where, "dateRange", { isCustom: true });
-          } else if (target.where.dateRange === "all") {
-            this.$set(target.where, "dateRange", { all: true });
-          }
-        }
-        if (!target.where.hasOwnProperty("DeviceId")) {
-          this.$set(target.where, "DeviceId", []);
-        }
-        if (!target.where.duration) {
-          this.$set(target.where, "duration", {});
-        }
-        if (target.where.DeviceId && target.where.DeviceGroups) {
-          target.where.DeviceId = [
-            ...target.where.DeviceId,
-            ...target.where.DeviceGroups
-          ];
-        } else if (target.where.DeviceGroups) {
-          target.where.DeviceId = [...target.where.DeviceGroups];
-        }
+    },
+    makeArray(value) {
+      if (typeof value === "object") {
```

Comment:
You might want `Array.isArray` here? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray